### PR TITLE
Update to v4.5.4

### PR DESCRIPTION
**Pull request by xylar**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
Seeing this error:
```
CMake Error at CMakeLists.txt:428 (ADD_EXECUTABLE):
  Cannot find source file:

    f90tst_parallel_compressed.f90

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc
Call Stack (most recent call first):
  nf03_test4/CMakeLists.txt:51 (build_bin_test)


CMake Error at CMakeLists.txt:428 (ADD_EXECUTABLE):
  No SOURCES given to target: f90tst_parallel_compressed
Call Stack (most recent call first):
  nf03_test4/CMakeLists.txt:51 (build_bin_test)
```
It seems like the confusion is because the file extension is `F90`, not `f90`.  @bekozi or @ocefpaf, might the fix be as simple as a patch to fix the file extension in `CMakeLists.txt`?  Or am I missing something.  My fluency with CMake could be classified as "very rudimentary" at best.

https://github.com/Unidata/netcdf-fortran/blob/v4.5.4/nf03_test4/CMakeLists.txt#L51

File: recipe/0002-fix-test-file-extension.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+diff -ruN netcdf-fortran-4.5.4/nf03_test4/CMakeLists.txt netcdf-fortran-4.5.4-patch/nf03_test4/CMakeLists.txt
+--- netcdf-fortran-4.5.4/nf03_test4/CMakeLists.txt	2022-01-07 14:43:57.000000000 -0700
++++ netcdf-fortran-4.5.4-patch/nf03_test4/CMakeLists.txt	2022-01-19 16:56:26.078419044 -0700
+@@ -48,7 +48,7 @@
+     build_bin_test(f90tst_parallel3 ".F90")
+     build_bin_test(f90tst_nc4_par ".F90")
+     build_bin_test(f90tst_parallel_fill ".f90")
+-    build_bin_test(f90tst_parallel_compressed ".f90")
++    build_bin_test(f90tst_parallel_compressed ".F90")
```
Comment (xylar) [inline recipe/0002-fix-test-file-extension.patch:9]: Fingers crossed that this is all it takes.

Comment (xylar) [reply recipe/0002-fix-test-file-extension.patch:9]: That seems to have done it.

**Comment by xylar** [comment]
@bekozi and @ocefpaf, please give this a quick look.  Feel free to merge when ready.  Hopefully, someone with more experience/interest will take on the Windows build.

File: recipe/0002-fix-test-file-extension.patch (hunk 1 above)
Comment (ocefpaf) [reply recipe/0002-fix-test-file-extension.patch:9]: That was an odd one :-)

**Pull request merged**